consensus/scp/tests/mock_network/scp_network.rs

```diff
@@ -0,0 +1,178 @@
+//! A simulated SCP network.
+
+use crate::mock_network::{
+    scp_node::{SCPNode, SCPNodeSharedData},
+    NodeConfig, TestOptions,
+};
+use mc_common::{
+    logger::{log, Logger},
+    NodeID,
+};
+use mc_consensus_scp::Msg;
+use std::{
+    collections::{HashMap, HashSet},
+    sync::{Arc, Mutex},
+    thread::JoinHandle,
+};
+
+pub struct SCPNetwork {
+    /// NodeID to node.
+    nodes: Arc<Mutex<HashMap<NodeID, SCPNode>>>,
+
+    /// NodeID to the node's thread handle.
+    handles: HashMap<NodeID, JoinHandle<()>>,
+
+    /// NodeID to node name.
+    pub names: HashMap<NodeID, String>,
+
+    /// NodeID to ...???
```

Comment:
`SCPNodeSharedData` is basically just a mock ledger and could probably be cleaned up to that effect.